rust-mode.el

```diff
@@ -51,6 +51,8 @@
 ;;; Start of a Rust item
 (defvar rust-top-item-beg-re
   (concat "\\s-*\\(?:priv\\|pub\\)?\\s-*"
+          "\\(:?([[:word:]_[:digit:]:]+)\\)?\\s-*"
+          "\\(:?unsafe\\)?\\s-*"
```

Comment:
Are these supposed to be shy groups? The syntax there is `?:`, not `:?`.

Rather than using `[[:word:]_[:digit:]:]+`, I think it would be better to say `\_+(::\_+)*`. This uses the `\_` "symbol constituent" syntax class.

Reply:
Uhh, I got the syntax for symbol constituents wrong. But this should probably use `rust-re-ident`.